### PR TITLE
Remove Spree::Store

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj it looks like this removes the file `core/app/views/spree/shared/_base_mailer_header.html.erb` yet it looks like there's still a reference to it in https://github.com/MountainRoseHerbs/spree/blob/master/core/app/views/layouts/spree/base_mailer.html.erb#L756

Also, possibly OT, but the footer is very similar to this in that it's completely commented out. I'd recommend we remove both the header and footer, and references to them. Ideally we'd remove the whole `base_mailer.html.erb` altogether, but I can understand that being done in a later PR.

**Comment by dkubb** [comment]
It looks like there may be some references to the old api stores controller, eg:

```
$ ag -i 'store' */config/rubocop.yml
api/config/rubocop.yml
72:    - 'app/controllers/spree/api/stores_controller.rb'
131:    - 'spec/controllers/spree/api/stores_controller_spec.rb'
167:    - 'spec/controllers/spree/api/stores_controller_spec.rb'

frontend/config/rubocop.yml
92:    - 'app/controllers/spree/store_controller.rb'
100:    - 'app/controllers/spree/store_controller.rb'
121:    - 'app/controllers/spree/store_controller.rb
```

**Comment by dkubb** [comment]
I also found this:

```
$ ag 'StoreController'
frontend/app/controllers/spree/home_controller.rb
2:  class HomeController < Spree::StoreController

frontend/app/controllers/spree/checkout_controller.rb
13:  class CheckoutController < Spree::StoreController

frontend/app/controllers/spree/orders_controller.rb
2:  class OrdersController < Spree::StoreController

frontend/app/controllers/spree/store_controller.rb
2:  class StoreController < Spree::BaseController

frontend/spec/controllers/spree/current_order_tracking_spec.rb
4:  controller(Spree::StoreController) do
```

**Comment by mbj** [comment]
> I realize many of these need to remain, since they are used by other parts of the system and they aren't strictly related to Spree::Store, but I figured I'd flag them just in case.

All of the `StoreControllers` need to remain in place for now as its a base class for frontend controllers. Not a good naming choice, but nothing I do want to fix this iteration.

**Comment by mbj** [comment]
> . Ideally we'd remove the whole base_mailer.html.erb altogether, but I can understand that being done in a later PR.

Actually we removed all HTML email views, so this layout is just a leftover. I could remove it in this PR.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

File: frontend/spec/support/order_walkthrough.rb [hunk 1]
```diff
@@ -22,10 +22,7 @@ def self.up_to(shipping_method, state)
 
       zone.members.create(zoneable: country)
 
-      order = Spree::Order.create!(
-        store: Spree::Store.default,
-        email: 'spree@example.com'
-      )
+      order = Spree::Order.create!(email: 'spree@example.com')
```
Comment (mbj) [inline frontend/spec/support/order_walkthrough.rb]: Yeah, good catch. I'll do that change.

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

**Pull request merged**